WordPress/gutenberg
# [PoC] Patterns: try partial syncing of inner blocks

## What?
This is an exploration to see if it will be possible to partially sync a block that has innerBlocks as part of the work around https://github.com/WordPress/gutenberg/issues/53705.

## Why?
The initial work around partial syncing is focusing on simple block content like a paragraph's content attribute or an image's src. but we need to make sure that the APIs being used to this will scale to allowing blocks with innerBlocks content to also be partially synced.

## How?
Uses the handling of the patterns innerBlocks implemented in #56235 to detect changes to a patterns inner blocks and save the serialized changes into the patterns attributes

## Testing Instructions

- Currently only works with List block
- Enable Connections experiment
- Create a synced pattern and add a list block
- In the Connections panel connect the list to `Pattern attributes`
- Add the pattern to a post and edit the list items and make sure changes display correctly in frontend

This is a very rough prototype, and will most likely have bugs 😄  

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/3629020/6e88afb5-07ed-45ba-af43-2a0565dfdc26
